metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/stellar/GetProfile.java

```diff
@@ -102,94 +90,108 @@
         returns="The selected profile measurements."
 )
 public class GetProfile implements StellarFunction {
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
   /**
-   * Cached client that can retrieve profile values.
+   * Allows the function to retrieve persisted {@link ProfileMeasurement} values.
    */
-  private ProfilerClient client;
+  private ProfilerClient profilerClient;
 
   /**
-   * Cached value of config map actually used to construct the previously cached client.
+   * Creates the {@link ProfilerClient} used by this function.
    */
-  private Map<String, Object> cachedConfigMap = new HashMap<String, Object>(6);
-
-  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private ProfilerClientFactory profilerClientFactory;
 
   /**
-   * Initialization.  No longer need to do anything in initialization,
-   * as all setup is done lazily and cached.
+   * Last known global configuration used to create the {@link ProfilerClient}. If the
+   * global configuration changes, a new {@link ProfilerClient} needs to be constructed.
    */
+  private Map<String, Object> lastKnownGlobals = new HashMap<>();
+
+  public GetProfile() {
+    profilerClientFactory = new HBaseProfilerClientFactory();
+  }
+
   @Override
   public void initialize(Context context) {
+    Map<String, Object> globals = getGlobals(context);
+    profilerClient = profilerClientFactory.create(globals);
   }
 
-  /**
-   * Is the function initialized?
-   */
   @Override
   public boolean isInitialized() {
-    return true;
+    return profilerClient != null;
   }
 
-  /**
-   * Apply the function.
-   * @param args The function arguments.
-   * @param context
-   */
   @Override
-  public Object apply(List<Object> args, Context context) throws ParseException {
+  public void close() throws IOException {
+    if(profilerClient != null) {
+      profilerClient.close();
+    }
+  }
 
+  @Override
+  public Object apply(List<Object> args, Context context) throws ParseException {
+    // required arguments
     String profile = getArg(0, String.class, args);
     String entity = getArg(1, String.class, args);
-    Optional<List<ProfilePeriod>> periods = Optional.ofNullable(getArg(2, List.class, args));
-    //Optional arguments
```

Comment:
Moved all of this logic around getting optional args into `getGroups` and `getOverrides`.